### DISCUSSION TITLE
Examples and Test Things Management

### DISCUSSION THREAD

**Discussion by egekorkan** [eclipse-thingweb/thingweb#3] [General]
Currently, the way we manage the live Things in our Eclipse server is very suboptimal. They are started in different "screens" and when they crash we do not know that they do. Additionally, with the test-things repo, we are creating Things for testing the protocols. The question is how do we manage all of these. My idea is to pack them together to one docker compose file (see https://github.com/eclipse-thingweb/test-things/blob/main/docker-compose.yml) and add some observation tools on top. This raises the issue of synchronization of these Things and node-wot codebase.

Some points to not forget:
- examples are used for build tests in node-wot
- in events like plugfests or demos, we need to quickly update the deployed Things. So merging a PR on the master of node-wot and redeploying is useful.

We can do one of the following:
- Moving all examples to test-things and having real tests in node-wot
- Moving only the dist to test-things (node-wot keeps src but dist are generated in test-things)
- Creating an "examples" npm package from node-wot and reusing it in test-things

@danielpeintner @relu91 and others, feel free to comment for an option

**Comment by danielpeintner** [comment]
I think it still makes sense to have *some* examples in node-wot while we can move most of the examples to test-things. Having said that, we should link test-things from node-wot also so that people find more (sophisticated) examples.

Anyhow, removing all examples from node-wot seems seem wrong to me since we don't want to push people getting started with node-wot to clone yet another repo etc.
